cylc/flow/run_modes/skip.py

```diff
@@ -0,0 +1,161 @@
+# THIS FILE IS PART OF THE CYLC WORKFLOW ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Utilities supporting skip modes
+"""
+from logging import INFO
+from typing import (
+    TYPE_CHECKING, Dict, List, Tuple)
+
+from cylc.flow.exceptions import WorkflowConfigError
+from cylc.flow.task_outputs import (
+    TASK_OUTPUT_SUBMITTED,
+    TASK_OUTPUT_SUCCEEDED,
+    TASK_OUTPUT_FAILED,
+    TASK_OUTPUT_STARTED
+)
+from cylc.flow.run_modes import RunMode
+
+if TYPE_CHECKING:
+    from cylc.flow.taskdef import TaskDef
+    from cylc.flow.task_job_mgr import TaskJobManager
+    from cylc.flow.task_proxy import TaskProxy
+    from typing_extensions import Literal
+
+
+def submit_task_job(
+    task_job_mgr: 'TaskJobManager',
+    itask: 'TaskProxy',
+    rtconfig: Dict,
+    _workflow: str,
+    now: Tuple[float, str]
+) -> 'Literal[True]':
+    """Submit a task in skip mode.
+
+    Returns:
+        True - indicating that TaskJobManager need take no further action.
+    """
+    # Don't do anything if task is held:
+    if itask.state.is_held:
+        return True
```

Comment:
I don't think this bit is right (for any of the non-live modes).

The `is_held` flag is respected at the queue level:

https://github.com/cylc/cylc-flow/blob/87bb2d49a53c7807d432b3dbb47b679a0b275796/cylc/flow/task_queues/independent.py#L57-L62

It works by holding tasks back from being submitted, so the submission logic itself doesn't have to implement this logic.

This is causing an issue because `cylc trigger` should run a task even if it is held, but at the moment, held tasks cannot be triggered e.g. in skip mode whereas they can in live mode.

E.G:

```
cylc vip --pause
cylc hold '*'
cylc play
cylc trigger '//*'
```